### PR TITLE
No Null Embeddings

### PR DESCRIPTION
**Pull request by yuhongsun96**
## Description
Rework logic to handle documents that have no title or no content. Stricter enforcement of requirements for embedding

## How Has This Been Tested?
Indexed documents that have no title or no content to ensure they were dropped.
Indexed normals docs

## Accepted Risk
No full QA process done, just tried to reproduce the edge cases

## Related Issue(s)
NA

## Checklist:
- [x] All of the automated tests pass
- [x] All PR comments are addressed and marked resolved
- [x] If there are migrations, they have been rebased to latest main
- [x] If there are new dependencies, they are added to the requirements
- [x] If there are new environment variables, they are added to all of the deployment methods
- [x] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [x] Docker images build and basic functionalities work
- [x] Author has done a final read through of the PR right before merge

### CODE REVIEW

File: backend/danswer/indexing/chunker.py [hunk 1]
```diff
@@ -280,14 +256,17 @@ def chunk_document(
...
+                    else None,
                 )
             )
             link_offsets = {0: section_link_text}
             chunk_text = section_text
 
-    # Once we hit the end, if we're still in the process of building a chunk, add what we have
-    # NOTE: if it's just whitespace, ignore it.
-    if chunk_text.strip():
+    # Once we hit the end, if we're still in the process of building a chunk, add what we have. If the is only whitespace left
```
Comment (hagen-danswer) [inline backend/danswer/indexing/chunker.py]: Nit: typo
